### DISCUSSION TITLE
JS solution not working, although is correct and working locally.

### DISCUSSION THREAD

**Discussion by michskrzpriv** [codewars/codewars.com#2787] [Q&A] [answered]
There's an easy fundamental javascript task:
_An isogram is a word that has no repeating letters, consecutive or non-consecutive. Implement a function that determines whether a string that contains only letters is an isogram. Assume the empty string is an isogram. Ignore letter case.

**Example: (Input --> Output)_

"Dermatoglyphics" --> true
"aba" --> false
"moOse" --> false (ignore letter case)**

My solution is:

```
function isIsogram(str){
  let counts = {};
  counts = str.split('').map( elem => !counts[elem] ? counts[elem] = 1 : counts[elem] += 1 );

  return Math.max( ...counts ) === 1;
}
```

However it doesn't work within the Kata test environment. When I tested it locally and using jsfiddle.net it does work! I did some research within the code, and it looks like counts[elem] = 1 does not work when elem is a variable.

In other words, you can define 

`counts['m'] = 1`

but 
```
let m = 'm';
counts[m] = 1;
```

This seems like some Kata testing environment bug. But please correct me if I'm wrong.l

**Comment by hobovsky** [comment, marked as answer]
What does your solution return for "moOse", or "KEep"? It should return `false` for both.

Please mind that Github boards are not a good place to discuss questions related to.solving kata. The best place are kata discourse, or Codewars Discord and `#help-solve` channel.

Cheers!

**Reply by michskrzpriv** [reply]
You're right.

**Comment by michskrzpriv** [comment]
Forget it. My Bad. Sorry

**Comment by michskrzpriv** [comment]
This thread can be deleted :)
